### PR TITLE
Implement missing permission checks

### PR DESCRIPTION
**Pull request by beganovich**
This implements a few missing permission checks for the app.

### CODE REVIEW

File: src/components/CompanySwitcher.tsx [hunk 1]
```diff
@@ -38,16 +39,15 @@ export function CompanySwitcher() {
 
   const state = useSelector((state: RootState) => state.companyUsers);
 
-  const canUserAddCompany = isSelfHosted() || (isHosted() && !freePlan());
+  const companyUser = useCurrentCompanyUser();
 
-  const dispatch = useDispatch();
+  const canUserAddCompany =
+    (isSelfHosted() || (isHosted() && !freePlan())) && companyUser?.is_admin; // @Todo: Needs confirmation.
```
Comment (beganovich) [inline src/components/CompanySwitcher.tsx:45]: @turbo124 for confirmation.

File: src/pages/dashboard/components/Totals.tsx [hunk 2]
```diff
@@ -97,6 +99,11 @@ export function Totals() {
   };
 
   const getTotals = () => {
+    if (!hasPermission('view_payment')) {
+      // @Todo: Correct permission for fetching totals.
```
Comment (beganovich) [inline src/pages/dashboard/components/Totals.tsx:103]: @turbo124 for confirmation.

File: src/pages/dashboard/components/Totals.tsx [hunk 3]
```diff
@@ -114,6 +121,11 @@ export function Totals() {
   };
 
   const getChartData = () => {
+    if (!hasPermission('view_payment')) {
+      // @Todo: Correct permission for fetching totals.
```
Comment (beganovich) [inline src/pages/dashboard/components/Totals.tsx:125]: @turbo124 for confirmation.

File: src/pages/purchase-orders/routes.tsx [hunk 4]
```diff
@@ -17,12 +19,52 @@ import { Pdf } from './pdf/Pdf';
 
 export const purchaseOrderRoutes = (
   <Route path="/purchase_orders">
-    <Route path="" element={<PurchaseOrders />} />
+    <Route
+      path=""
+      element={
+        <Guard
+          guards={[() => permission('view_purchase_order')]} // @Todo: Missing bitmask guard
```
Comment (beganovich) [inline src/pages/purchase-orders/routes.tsx:26]: @turbo124 for bitmask int.

**Review by beganovich — COMMENTED** [review]
Few things required before merge.

**Comment by turbo124** [comment]
public const MODULE_PURCHASE_ORDERS = 16384;

File: src/components/CompanySwitcher.tsx (hunk 1 above)
Comment (turbo124) [reply src/components/CompanySwitcher.tsx:45]: company_user.is_owner == true

only

File: src/pages/dashboard/components/Totals.tsx (hunk 2 above)
Comment (turbo124) [reply src/pages/dashboard/components/Totals.tsx:103]: is_admin || is_owner

File: src/pages/dashboard/components/Totals.tsx (hunk 3 above)
Comment (turbo124) [reply src/pages/dashboard/components/Totals.tsx:125]: is_admin || is_owner

File: src/common/hooks/permissions/useHasPermission.ts [hunk 5]
```diff
@@ -30,12 +29,17 @@ type RecurringExpensePermissions =
   | 'create_recurring_expense'
   | 'view_recurring_expense'
   | 'edit_recurring_expense';
-type BankTransactionsPermissions =
+type BankTransactionPermissions =
   | 'create_transaction'
   | 'view_transaction'
   | 'edit_transaction';
```
Comment (turbo124) [inline src/common/hooks/permissions/useHasPermission.ts:35]: please prefix with bank_transaction instead of transaction

**Review by turbo124 — CHANGES_REQUESTED** [review]
as a user not with transaction permissions, i am still seeing the transactions tab?

also, the transactions and recurring expenses are not appearing in the permissions list for me.

**Comment by beganovich** [comment]
@turbo124 for revivew.

**Comment by turbo124** [comment]
@beganovich testing on this now looks good!

**Comment by beganovich** [comment]
Closing in favor of #464 .

**Pull request closed without merging**